### ISSUE TITLE
rgeeExtra:  An extensions for 'rgee'

### ISSUE THREAD

**Issue by csaybar** [ropensci/software-review#608]
Submitting Author Name: Cesar Aybar
Submitting Author Github Handle: @csaybar
Other Package Authors Github handles: (comma separated, delete if none) @davemlz, @LBautistaB13, @choisy
Repository:  https://github.com/r-earthengine/rgeeExtra
Version submitted: 0.0.1 
Submission type: Standard
Editor: @maurolepore
Reviewers: @nmorandeira, @zackarno

Archive: TBD
Version accepted: TBD
Language: en

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: rgeeExtra
Title: Extensions for 'rgee'
Version: 0.0.1
Authors@R: 
    c(person(given = "Cesar",
           family = "Aybar",
           role = c("aut", "cre"),
           email = "csaybar@gmail.com",
           comment = c(ORCID = "0000-0003-2745-9535")),
      person(given = "David",
           family = "Montero",
	         role = c("ctb"),
	         email = "dml.mont@gmail.com",
	         comment = c(ORCID = "0000-0002-9010-3286")),
      person(given = "Lesly",
           family = "Bautista",
	         role = c("ctb"),
	         email = "leslyarcelly.213@gmail.com",
	         comment = c(ORCID = "0000-0003-3523-8687")),
      person(given = "Marc",
           family = "Choisy",
	         role = c("ctb"),
	         email = "mchoisy@oucru.org",
	         comment = c(ORCID = "0000-0002-5187-6390"))
    )
Description: High-level API to process Google Earth Engine (GEE) raster (ee.Image and ee.ImageCollection) and vector data (ee.Geometry, ee.Feature, and ee.FeatureCollection). Popular Third-party GEE algorithms are re-coded from Javascript and Python to R.
License: Apache License (>= 2)
Encoding: UTF-8
LazyData: true
Roxygen: list(markdown = TRUE)
RoxygenNote: 7.2.3
Config/reticulate:
  list(
    packages = list(
      list(package = "earthengine-api")
    )
  )
Imports:
    reticulate (>= 1.31),
    rgee,
    magrittr,
    crayon,
    cli
Suggests:
    testthat (>= 3.0.0),
    googledrive,
    googleCloudStorageR,
    geojsonio,
    magick,
    stars,
    sf,
    knitr,
    rmarkdown
VignetteBuilder: knitr
Language: en-US
BugReports: https://github.com/r-earthengine/rgeeExtra/issues
URL: https://github.com/r-earthengine/rgeeExtra
```

## Scope

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below. If you are unsure, we suggest you make a pre-submission inquiry.):

	- [x] data retrieval
	- [ ] data extraction
	- [ ] data munging
	- [ ] data deposition
    - [ ] data validation and testing
	- [ ] workflow automation
	- [ ] version control
	- [ ] citation management and bibliometrics
	- [ ] scientific software wrappers
	- [ ] field and lab reproducibility tools
	- [ ] database software bindings
	- [x] geospatial data
	- [ ] text analysis

- Explain how and why the package falls under these categories (briefly, 1-2 sentences):

[rgeeExtra](https://github.com/r-earthengine/rgeeExtra) is an extension of the [rgee](https://github.com/r-earthengine/rgee) package that enables interaction with [Google Earth Engine](https://earthengine.google.com/) (GEE). The advantage of rgeeExtra lies in its user-friendly syntax for R users. The package includes operator overloading and access to various functions created by third parties within the [ee_extra](https://github.com/r-earthengine/ee_extra) package.

-   Who is the target audience and what are scientific applications of this package?

GEE users and any R user who wants to retrieve data using the GEE API from within R.

-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?

There are other packages for connecting to GEE, but they are primarily focused on composite generation and data download. See [geedim](https://github.com/leftfield-geospatial/geedim)

-   (If applicable) Does your package comply with our [guidance around _Ethics, Data Privacy and Human Subjects Research_](https://devguide.ropensci.org/policies.html#ethics-data-privacy-and-human-subjects-research)?

Yes

-   If you made a pre-submission inquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

-   Explain reasons for any [`pkgcheck` items](https://docs.ropensci.org/pkgcheck/) which your package is unable to pass.

-  Package uses global assignment operator ('<<-'). The global assignment operator is used in the .onLoad special function to set the *ee_extra* Python Package in the R global environment (See [here](https://github.com/r-earthengine/rgeeExtra/blob/master/R/rgeeExtra-package.R)). This approach is similar to other reticulate R packages like keras or tensorflow.

## Technical checks

Confirm each of the following by checking the box.

- [x] I have read the [rOpenSci packaging guide](https://devguide.ropensci.org/building.html).
- [x] I have read the [author guide](https://devdevguide.netlify.app/authors-guide.html) and I expect to maintain this package for at least 2 years or to find a replacement.

This package:

- [x] does not violate the Terms of Service of any service it interacts with.
- [ ] has a CRAN and OSI accepted license.
- [x] contains a [README with instructions for installing the development version](https://ropensci.github.io/dev_guide/building.html#readme).
- [x] includes [documentation with examples for all functions, created with roxygen2](https://ropensci.github.io/dev_guide/building.html#documentation).
- [x] contains a vignette with examples of its essential functions and uses.
- [x] has a [test suite](https://ropensci.github.io/dev_guide/building.html#testing).
- [x] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage.

## Publication options

- [ ] Do you intend for this package to go on CRAN?
- [ ] Do you intend for this package to go on Bioconductor?
- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:

<details>
<summary>MEE Options</summary>

- [ ] The package is novel and will be of interest to the broad readership of the journal.
- [ ] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [x] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.org/code-of-conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by csaybar** [comment]
@ropensci-review-bot

**Comment by maurolepore** [comment]
@csaybar thanks a lot for your submission. 

* Can you confirm you're requesting a review in English?
* Also, this is part of the Champions program, right?

`@mpadge` kindly noted that coverage and R CMD check failed due to a missing python dependency that our system can't track. I'll need a bit of time to investigate more. Do you know a good/easy way to provide the required Python environment so that the handling editor and reviewers can hit the ground running?

**Comment by csaybar** [comment]
Hi @maurolepore thank you for your quick response. I just activated the notification button to this issue.

*Can you confirm you're requesting a review in English?*

Yes, I'd prefer to do it in English

Also, this is part of the Champions program, right?

Yes, I'm part of the program and my mentor is Marc Choisy @choisy. 

Unfortunately, I haven't been able to resolve the issues related to Python dependencies. These are the same ones you can find in the [pangoling](https://github.com/bnicenboim/pangoling) package review. The Python coverage when it is evaluated in an ENV with the Python envs is around 95%. See https://app.codecov.io/gh/csaybar/rgeeExtra/tree/master/R

**Comment by csaybar** [comment]
@ropensci-review-bot

**Comment by maurolepore** [comment]
Great, thanks for pointing me to pangoling. I'll explore how they handled installation during [review](https://github.com/ropensci/software-review/issues/575).

**Comment by maurolepore** [comment]
@ropensci-review-bot assign @maurolepore as editor

**Comment by maurolepore** [comment]
@csaybar, I self-assigned to be the handling editor. I think this makes sense because I would like to explore the installation process, I'm part of the Champions program, and I'm available. 

I'll start working on the handling-editor's checklist. In the meantime, could you please list three potential reviewers? 

Following our guidelines I'll use no more than one of them, but it's useful to  have a few options and also to see the range of expertise you think would add value to this submission.

**Comment by csaybar** [comment]
Hi, @maurolepore, 

Here it is.

@davemlz, @giswqs, @yabellini, @ricds

**Comment by maurolepore** [comment]
@csaybar, 

In the installation instructions of rgee I see you point to a [posit.cloud project](https://posit.cloud/content/5175749) with a file that aims to quickly help a new user set everything up in a fresh environment. That seems like a good idea. Could you please do ti also for rgeeExtra, so that I and the reviewers can install the package as quickly as possible?

Note however that instructions for rgee don't seem to just work on posit.cloud as I seem to need `gcloud` and I can't use `sudo` to install it on posit.cloud. [Here is the project I used](https://posit.cloud/content/6633813). You'll see some FIXME that explain my struggles. As rgeeExtra depends on rgee I guess simplifying the installation of rgee will also help with rgeeExtra?

If posit.cloud limits the system dependencies we can install, how else can we quickly create a working environment for the reviewers? Would it help to have a Docker image on r-hub or a Dockerfile in the repo?

**Comment by csaybar** [comment]
Hi @maurolepore,

I will prepare a reproducible demo in posit/colab over the weekend to facilitate the review process.

I will be sure to keep you updated.

**Comment by csaybar** [comment]
Hi @maurolepore 

Here is an example of how to use rgeeExtra in posit cloud.
https://posit.cloud/content/6670766

It is important that the rgeeExtra user has a **GEE account**.

Unfortunately, positCloud does not have a way to send the token back (I tried to solve this problem a year ago). Kevin Ushey, explain this [here](https://twitter.com/csaybar/status/1576427594383519744)

"It's an unfortunate limitation of system() in RStudio; we currently don't have a way of routing stdin from the R process to subprocesses. You could instead run this in a separate Terminal instance for now."

![image](https://github.com/ropensci/software-review/assets/16768318/441216d4-25d0-42c1-831f-580bc91b6de6)

So the rgee user who wants to use PositCloud should do so through the terminal using: 

```
$ earthengine authenticate --auth_mode notebook
```

![image](https://github.com/ropensci/software-review/assets/16768318/df37597e-6a8d-4f3e-8474-55c0d88f34ec)


After that rgee & rgeeExtral will work perfectly! I attached two examples that show rgeeExtra features.

![image](https://github.com/ropensci/software-review/assets/16768318/7f27f0d8-4407-4791-90ab-abb7500bb8fd)

Feel free to reach out if there's anything I can assist you with!

**Comment by maurolepore** [comment]
Thanks @csaybar, 

I tried your instructions for installation and usage a few times. I arrived to the version below that seems succinct and functional. What do you think? 

After your review it would be nice to edit README to ensure the smoothest experience for anyone trying to install and use the package.

----

## Installation

```r
install.packages("remotes")
remotes::install_github("r-spatial/rgee")
remotes::install_github("r-earthengine/rgeeExtra")
rgee::ee_install()
```

Authenticate with Google Earth Engine

```bash
# Run in the Terminal (not in the R console)
earthengine authenticate --auth_mode notebook
```

## Example

```r
library(rgee)
library(rgeeExtra)

ee_Initialize()
extra_Initialize()

img <- ee$Image("LANDSAT/LC08/C01/T1_SR/LC08_038029_20180810")
ndvi <- (img[["B5"]] - img[["B4"]])/(img[["B5"]] + img[["B4"]])**2
names(ndvi) <- "pow_ndvi"

Map$centerObject(img,zoom=12)
Map$addLayer(img)
```

---

Now that I can install the package I will continue with all other editor checks.

**Comment by maurolepore** [comment]
@ropensci-review-bot check

**Comment by maurolepore** [comment]
@ropensci-review-bot check package

**Comment by csaybar** [comment]
Hi @maurolepore, we know that setting up an environment with both Python and R can be a bit tiring. Unfortunately, there's not much we can do from our end since the connection relies on reticulate. I've updated the README for rgeeExtra and included a link that explains how to install rgee Python dependencies. Hope it can help.

**Comment by maurolepore** [comment]
### Editor checks:

- [ ] **Documentation**: The package has sufficient documentation available online (README, pkgdown docs) to allow for an assessment of functionality and scope without installing the package. In particular,
    - [ ] Is the case for the package well made?
    - [ ] Is the reference index page clear (grouped by topic if necessary)?
    - [ ] Are vignettes readable, sufficiently detailed and not just perfunctory?
- [ ] **Fit**: The package meets criteria for [fit](https://devguide.ropensci.org/policies.html#package-categories) and [overlap](https://devguide.ropensci.org/policies.html#overlap).
- [ ] **Installation instructions:** Are installation instructions clear enough for human users?
- [ ] **Tests**: If the package has some interactivity / HTTP / plot production etc. are the tests using [state-of-the-art tooling](https://devguide.ropensci.org/building.html#testing)?
- [ ] **Contributing information**: Is the documentation for contribution clear enough e.g. tokens for tests, playgrounds?
- [ ] **License:** The package has a CRAN or OSI accepted license.
- [ ] **Project management**: Are the issue and PR trackers in a good shape, e.g. are there outstanding bugs, is it clear when feature requests are meant to be tackled?
---

#### Editor comments

@csaybar,

To help you track my comments, I'll label them with "ml" and a numerical sequence: ml01, ml02, etc:

* Comments starting with "bullet" (like this one) suggest things that do not require your action or response.

- [ ]  Comments starting with "checkbox" (like this one) ask for things that require your action and response.
 
 
--

**The package has sufficient documentation available online (README, pkgdown docs) to allow for an assessment of functionality and scope without installing the package.**

- [ ] ml01. In all code examples please show not only the code but also its outout. This will help the reviewers and users to assess the package functionality without installing the package and running the code. 

You may evaluate code chunks in .Rmd or .qmd documents or with the [reprex package](https://reprex.tidyverse.org/). It's best to use the option `comment = "#>"` so that the motivated reader can copy/paste/run without errors.

**Is the case for the package well made?**

- [ ] ml02. Please Answer "Why rgeeExtra is needed?" at the top of README. That location is the most useful for reviewers and new users.

Note the [output of `usethis::use_readme_rmd()`](https://github.com/r-lib/usethis/blob/main/inst/templates/package-README) starts with "The goal of {{{ Package }}} is to ...". 

Currently the top of README talks about GEE and rgee but not rgeeExtra ([What is Google Earth Engine?](https://github.com/r-earthengine/rgeeExtra#what-is-google-earth-engine)). The reader sees the [Installation or rgeeExtra](https://github.com/r-earthengine/rgeeExtra#installation) before knowing about it.

I think the best case for the package is made at top of the [Features article](https://r-earthengine.com/rgeeExtra/articles/Features.html). Moving this text to the top or README would solve the problem :-)

> rgeeExtra serves as a wrapper for the Python package named eeExtra. The creation of eeExtra was driven by a need to consolidate various third-party GEE packages and projects found on GitHub.

> rgeeExtra ensures a seamless integration of these projects across different programming languages by standardizing methods and facilitating the use of JavaScript code projects beyond the traditional Code Editor environment.

Also in other places I see text that can help make the case for the package:

https://r-earthengine.com/rgeeExtra/articles/Introduction.html:

> rgeeExtra is an extension of the rgee package, enhancing the capabilities of Google Earth Engine (GEE) integration within the R environment.

https://r-earthengine.com/rgeeExtra/index.html#how-does-it-work:

> The rgeeExtra extends the following Earth Engine classes: ...

In any case, all candidate text seems to answer "what is rgeeExtra" but to make a stronger case for the package the question that needs to be answered is "why is reeExtra needed?".

**Is the reference index page clear (grouped by topic if necessary)?**

- [ ] ml03. In the [reference index](https://r-earthengine.com/rgeeExtra/reference/index.html), please [group the functions by topic](https://devdevguide.netlify.app/pkg_building#function-grouping). This will help the reviewers and users to immediately see the structure of the public API. It may also help you find gaps or redundancy.

**Are vignettes readable, sufficiently detailed and not just perfunctory?**

* ml04. The content of the articles seems very useful, and relocating some of it can help to ensure the content reaches reviewers and users where they most commonly need it and expect it. For example:

    - Make the case for the package at the top of README (moving text from the article "Features" as suggested above).
    - Show installation only in README, before the first usage of the package, and ask multiple new users to confirm they can install and use the package by following the instructions in the exact order they are presented in the installation section in README. Also ensure they follow and give feedback on instructions linked to from README but outside of it. 
    - Turn the article [Introduction](https://r-earthengine.com/rgeeExtra/articles/Introduction.html) into the common "Get started" article (via `usethis::use_article("name-of-your-package")`). Remove the installation section and instead point to the installation in README. If you keep installation instructions, ensure they come before (not after) the first example of usage. Considering explaining the code in prose step by step, rather than showing a big code-chunk then explaining it all after the end. The final figure seems misplaced. It would be more natural to see a plot as the output of a code chunk -- so it should follow the code, not the prose. 
  
**Fit: The package meets criteria for fit and overlap.**

* ml05. The package fits nicely in the stated categories, but consider explaining more about the overlap with rgee. For inspiration see how the [targets package explains why it supersedes drake](https://books.ropensci.org/targets/drake.html).

**Installation instructions: Are installation instructions clear enough for human users?**

- [ ] mlxx. As is the installation instructions in README didn't work for me. After several iterations I arrived to
a this revision: https://github.com/ropensci/software-review/issues/608#issuecomment-1755823040. 
I suggest asking other newbies like me to try it before asking the reviewers to install it. The limited reviewer's time and
energe will be better spent in providing more interesting feedback.

**Tests: If the package has some interactivity / HTTP / plot production etc. are the tests using state-of-the-art tooling?

- [ ] ml06. As much as possible please move tests setup code from each test file to a single [setup.R file](https://r-pkgs.org/testing-design.html#testthat-setup-files), a test helper, or other [advanced testing techniques](https://r-pkgs.org/testing-advanced.html).

- [ ] ml07. Please help me understand why there are many more functions in the [reference index](https://r-earthengine.com/rgeeExtra/reference/index.html) than test-files? It's common for each `R/exported_function.R` to have a corresponding `test/testthat/test-exported_function.R`.

- [ ] ml08. Please address the R CMD check failure on GitHub. The last passing commit dates August 18, 2023: https://github.com/r-earthengine/rgeeExtra/commits/master. Do you need to fix some test, or maybe use `skip_on_ci()` or similar?

**Contributing Is: information the documentation for contribution clear enough e.g. tokens for tests, playgrounds?**

- [ ] ml09. Is there any place in the docs where you explain how a contributor or user gets an account with GEE and a token? If not, please add it to the installation instrucitons or wherever you think it's most needed.

**License:** The package has a CRAN or OSI accepted license.

Yes.

**Project management: Are the issue and PR trackers in a good shape, e.g. are there outstanding bugs, is it clear when feature requests are meant to be tackled?**

- [ ] ml10. Please review the open issues. The total number of issues is fairly small, and it seems you may be able to close some already (e.g. 4, 5). For the remaining open issues, it would be nice to see a comment where you explain if you plan to address them. It's important to show the reviewers that we've made as much progress on our own before asking their feedback.


**Also**

* Running the exmple in README throws:

```r
#> The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,
#> which was just loaded, were retired in October 2023.
#> Please refer to R-spatial evolution reports for details, especially
#> https://r-spatial.org/r/2023/05/15/evolution4.html.
#> It may be desirable to make the sf package available;
#> package maintainers should consider adding sf to Suggests:.
```

 I'm puzzled. Both rgee and rgeeExtra are alraedy using sf. Do you know where this is comming from?

**Comment by maurolepore** [comment]
@csaybar, 

Thanks for your patience. My feedback took time but I hope it will also save reviewers time later. Please address my comments before I start searching for reviewers.

**Comment by csaybar** [comment]
Hi @maurolepore, I apologize for the delayed response, and I quite appreciate all your comments. I plan to begin addressing your feedback during the next week. Have a good weekend!

**Comment by csaybar** [comment]
Hi, @maurolepore! I'm pleased to inform you that a new version of RgeeExtra has been released (v0.1.0). We've made comprehensive improvements to the documentation and included numerous additional examples. Please find attached the responses addressing all your comments.  https://r-earthengine.com/rgeeExtra/
Demo here: https://posit.cloud/content/6670766

### m01

We have updated our example documentation to include both code and their corresponding outputs. This enhancement facilitates a better understanding and allows readers to easily copy, paste, and execute examples without any issues, aligning with the core idea of rgeeExtra, improving the user experience of R users when they have to use GEE APi.

### m02

The updated README now clearly explains the goal and the why of rgeeExtra at the top, providing more clarity to new users.

### m03
The support for R6 classes in roxygen2/pkgdown (reticulate) is still a bit buggy, we have been trying some workarounds, and now the method classes are available in the reference index. 



### m06
The installation instructions in the README have been revised and tested in Windows and Linux OS, ensuring they are clear and functional for all users. 

### m07/08/09
We have ensured that each function in the R/ folder has a corresponding test file in test/testthat/. We updated the codecov state: https://app.codecov.io/github/csaybar/rgeeExtra/tree/master 
All the 135 now pass successfully.

### m10

We include the following line in the installation section: 'For additional details on connecting GEE and R, refer to the information provided here. To set up an account, click here.'"

### m11

We have made a lot of changes to the API in the past, and all the issues reported were associated with previous versions that are no longer supported. For the first time, we have released a stable version (v.0.1.0), and our intention from now is to maintain its stability respecting the API, we decided.

Regarding the R spatial legacy dependency issues, we currently have an indirect dependency on leafletExtra2. Unfortunately, this package has not yet initiated the transition to sf.

**Comment by maurolepore** [comment]
Thanks a lot @csaybar!

Over the next few days I'll review your response and move on to searching for reviewers.

**Comment by maurolepore** [comment]
Dear @csaybar, 

I'm exploring the potential reviewers you listed [above](https://github.com/ropensci/software-review/issues/608#issuecomment-1731436320) and realize I forgot to highlight that our policies about [conflict of interest for reviewers](https://devdevguide.netlify.app/softwarereview_policies.html#coi) (COI) might shrink the list. For example, [David's profile](https://github.com/davemlz) suggest a COI: "I'm the ... co-creator of [eeExtra](https://github.com/r-earthengine/ee_extra) and [easystac](https://github.com/cloudsen12/easystac) (with the incredible [@csaybar](https://github.com/csaybar))". If that contribution was within the last year, then it would be considered a COI. 

Can you please review the list with COI in mind? And while you're there, please briefly explain why you would value the revew of each one in your list? This is useful for me to sense what kind of experience you're looking for, so I can search for a similar profile elsewhere.

And if your list above already considers COI then great. I just need to confirm.

**Comment by csaybar** [comment]
Hi @maurolepore, my apologies for the confusion. I have updated the list in accordance with the COI and added a brief explanation. Please let me know if you have any questions or need further clarification. Thank you!

**Comment by maurolepore** [comment]
@ropensci-review-bot assign @nmorandeira as reviewer

**Comment by maurolepore** [comment]
@ropensci-review-bot assign @zackarno as reviewer

**Comment by maurolepore** [comment]
@ropensci-review-bot check package

**Comment by maelle** [comment]
@ropensci-review-bot help

**Comment by maurolepore** [comment]
@ropensci-review-bot help

**Comment by maurolepore** [comment]
@ropensci-review-bot check package

**Comment by maurolepore** [comment]
Hola @mdelfin,

Muchas gracias por tu interes in rgeeExtra y por tu feedback. Nota que esta pagina esta dedicada al proceso de revision de rOpenSci. 

Quizas estas buscando esta otra pagina?

https://github.com/r-earthengine/rgeeExtra/issues

**Comment by mdelfin** [comment]
Hola Mauro, la verdad estoy haciendo una app en shiny y con conexion a google earth engine con la liberia rgee, lo que mencionan en la parte de los comentarios, es la conexion con POSIT CLOUD, es ahi donde tengo el problema, no puedo conectar, ya que los pasos que mencionan de la termninal, esto esta todo muy bien, PERO al momento de relalizar el ejercicion con  ee_Initialize(), vuelve el mismo problema que menciona ustedes en la parte de arriba. Saludos

**Comment by maurolepore** [comment]
Happy new year everyone! 

@nmorandeira and @zackarno, I see your due date was inconveniently in the holiday period, so I think it's fair to expect your reviews a bit later than scheduled. Looking forward to them.

**Comment by maurolepore** [comment]
Feliz 2024 @mdelfin. Por favor nota que este no es el lugar para hacer preguntas sobre el paquete. Por favor dirigijte a https://github.com/r-earthengine/rgeeExtra/issues

**Comment by zackarno** [comment]
Happy new years! @maurolepore thanks ! I'm hoping/aiming to get it finished this week!

**Comment by zackarno** [comment]
Thanks for the fantastic package and the chance to review! I look forward to incorporating it into my work and exploring more use-cases over time!

I attempted to limit the scope of my review to just `{rgeeExtra}`, avoid  `{rgee}` functionality, and also not dig into the `EE_extra`/`EE-Mont` python packages which are wrapped. 

Below is my review, I tried to tag each commend with a `#uid` for ease of referencing.There is so much cool stuff here that I feel like I could keep digging for a while longer, but will have to stop myself here:

## Package Review

*Please check off boxes as applicable, and elaborate in comments below.  Your review is not limited to these topics, as described in the reviewer guide*

- **Briefly describe any working relationship you have (had) with the package authors.**
- [x] As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

## Documentation

The package includes all the following forms of documentation:

- [x]  **A statement of need:** clearly stating problems the software is designed to solve and its target audience in README
- [x]  **Installation instructions:** for the development version of package and any non-standard dependencies in README
  + all good, but I added something as a nice-to- have below!
- [ ]  **Vignette(s):** demonstrating major functionality that runs successfully locally
  +  see below issue/bugs found in vignettes that causes them not to run properly
- [ ] **Function Documentation:** for all exported functions
  +  see `Documentation - Functions` section
- [ ] **Examples:** (that run successfully locally) for all exported functions
  +  see `Documentation - Examples` section
- [x] **Community guidelines:** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

## Functionality

- [x] **Installation:** Installation succeeds as documented.
- [ ] **Functionality:** Any functional claims of the software have been confirmed.
  + See `Functionality - Functionality` section below
- [x] **Performance:** Any performance claims of the software have been confirmed.
- [x] **Automated tests:** Unit tests cover essential functions of the package and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines.

Estimated hours spent reviewing: 16 - probably longer than it should have given it was my first review.

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

## Review Comments

## General Comments

- One thing I found challenging about this package is that its not straight forward to get help with 
"?function_name" syntax. In `{rgee}` you are able to do it through `ee_help()` which is not as straight forward as ?function_name, but still better than nothing. Would there be a way to implement either ee_help() or ?function_name help here? #za_gc1

- It would be nice if the `How does it work` section of the readme explained a bit more about it's relation to `ee_extra`/`eemont`/`spectralIndex` python packages. #za_gc2
  + For most of this review I had thought you were just porting eemont functions over to rgeeExtra with reticulate, but at the end just realized that ee_extra (python) in the inst folder. Is that coming from: [ee_extra](https://github.com/r-earthengine/ee_extra) package somewhere in readme?

- It could be useful, especially for contributors, if the scope of the package was more clearly defined. The package naming makes it sounds like it could be a grab bag for all sorts of miscellaneous useful functions, but I don't think that is really the intent. Also as it is largely built in python, would you want to keep it that way or be open to more R development? I know that a lot of R development would be through `{rgee}` which wraps python, but it's still different than writing python to contribute to the `ee_extra` package - does that make sense? #za_gc3

- There are some nice defensive assertions you put in place on certain functions particularly the subletting functions. I think these really improve user experience and should be added more throughout the package. There are many cases where if the user inputs the wrong type of value the function will either a. eventually timeout, b. completely freeze/abort the session, c. give no warning and you only find out later. I've flagged a few in which some class assertions should help the user avoid this experience. #za_gc4

- `names()` - It seems likely you have already considered this, but I feel that it would make more sense if the output of `names()` was consistent across ImageCollection and Image classes. Currently names returns band names for Images and property names for ImageCollections. Why not just have `property_names()` and `names()` (or something like that) that works consistently since both Images an ImageCollections have both. #za_gc5

- would be nice to have a `list_spectral_indices()` type function (like eemont.listIndices())- would be cool if the object returned would provide the formula & reference for the index as well #za_gc6

- `Extra_closest()` - really cool function some comments (in addition to what I believe to be a small bug - see `Functionality-Functionality` section) #za_gc7

- Would it be possible to make it some functions allow the user to autocomplete arguments? I'm thinking of Extra_maskClouds, Extra_preprocess. #za_gc8


### Documentation - Installation instructions
 
- When I am using a more typical python workflow, I feel like I have a clear view of the venv and anything I'm adding to it. In this setup, it seems less visible. I believe that is largely intended and makes sense for a lot of R users. Nonetheless it would be nice if there was more documentation on this for people who are more comfortable with python. For example, I would personally find it useful to see an example of how to `pip install` all the necessary dependencies for both `{rgee}` and `{rgeeExtra}` into a pre-existing/or new `venv`. #za_di1

### Documentation - Examples 

- No documentation on exported mean method https://github.com/r-earthengine/rgeeExtra/blob/7ea67d064e56e9429971d64440d8424df68f6249/R/ops-math.R#L258-L264
- example provided for `Summary.ee.image.Image` returns an image with 0 pixels so I am not reall ysure what exactly it is supposed to be doing? #za_de1
- Example on exported `names.ee.feature.Feature` method doesn't show function being used: https://github.com/r-earthengine/rgeeExtra/blob/7ea67d064e56e9429971d64440d8424df68f6249/R/ops-subsetting-fc.R#L57-L69 #za_de2
- Missing roxygen examples for `[` `[[` examples in `ops-subsetting-image.R`. It's not entirely clear what these are meant to do on an image so I think examples help #za_de3
- `ops-subsetting-imagecollection` the 2 top funcs have the same roxygen description but doing different things #za_de4
- `ops-subsetting.R` top example should have a `library(rgeeExtra)` call right? #za_de5
- see output of `devtools::check()` #za_de6

### Documentation - Functions

- `ee$Image$Extra_preprocess()` - would be nice to have more information on how to use in examples, and documentation/vignettes. It's not totally clear what the `...` arguments should be as it references `ee$Image$cloudmask()` which I do not believe is a function (maybe you meant `Extra_maskClouds`?). Since users should set parameters differently depending `img`/`ic` perhaps some explanation of default values should be given. Examples and vignettes sort of give impression that user should run without modifying parameters. I guess defaults are set for `Copernicus/S2_SR`? #za_df1
- `Extra_matchHistogram()`-  Would be useful to add `Map$addLayer()|Map$addLayer()` to roxygen example on function #za_df2
- `ops-math` - roxygen in below snippet says multiplication, but shows addition #za_df3
https://github.com/r-earthengine/rgeeExtra/blob/7ea67d064e56e9429971d64440d8424df68f6249/R/ops-math.R#L40-L44
- `ee_utils_gif_save()` roxygen says `@return No return value, called to write a GIF file.` but it does return a logical after writing the gif (i'm not sure why this is necessary) #za_df4
- `Extra_matchHistogram()` could be nice to provide a link and or some information on matching method #za_df5

### Documentation - Vignettes

- [aditional vignette](https://github.com/r-earthengine/rgeeExtra/blob/master/vignettes/Aditional.Rmd#L33)
  + Most examples are missing the `Extra_` prefix on the functions so they result in error #dv1
  + spelling: "Aditional" -> "Additional"  :-) #dv2
- I would check other Vignettes for this same issue. #dv3
- Could be nice to have a quick reminder at top of vignettes that in order to follow along users should install `{leaflet.extras2}` #dv4

## Functionality/ User Experience

### Functionality -Installation

- Worked fine for me. However in the past I have seen users struggle (including myself), but I believe this has gotten easier. #fi1

### Functionality - Functionality

- `Exra_Closest() very handy function! but some issues/questions 
  + it returns and `ImageCollection` not an `Image` as written in the roxygen although, but based on how it is written to only return ImageCollection of 1 image, I think it should be converted to an Image? #za_ff1 (https://github.com/r-earthengine/rgeeExtra/blob/7ea67d064e56e9429971d64440d8424df68f6249/R/ee_ImageCollection.R#L18)
  + Can't take R formatted dates as written in roxygen (but should) #za_ff2

```{r}
library(rgee)
library(rgeeExtra)
library(tidyverse)
ee_Initialize()
extra_Initialize()

url_s2 <- "COPERNICUS/S2"
ic <- ee$ImageCollection(url_s2)

date_chr <- "2023-09-10"
date_dt<- as_date(date_chr)
ic_closest_from_dt_obj <-  ic$Extra_closest(date_dt)
#> Error in py_call_impl(callable, call_args$unnamed, call_args$named) : 
#>   ee.ee_exception.EEException: Invalid argument specified for ee.Date(): 2023-09-10
#> Run `reticulate::py_last_error()` for details.
```
  + Could not find any documentation on what is supposed to happen on geographically large ImageCollection multiple images on same date. Theoretically as a user I might think that it would give me an ImageCollection containing all the images closes to that date, but I don't think that is what you were going for. Nonetheless I think we need some instructions in the roxygen and vignettes. Also this below cases is an example where `Extra_closest()` silently fails and then crashes/aborts the session on the next call that tries to do something with the resulting object. This is a case where some assertions would improve user experience. #za_ff3
  
```{r}
ic_closest <-  ic$Extra_closest(date_chr) # works
class(ic_closest)
#> [1] "ee.imagecollection.ImageCollection" "ee.collection.Collection"           "ee.element.Element"                
#>[4] "ee.computedobject.ComputedObject"   "ee.encodable.Encodable"             "python.builtin.object"  

# if I run this in combination with above chunk my R session will freeze/abort
ee_get_date_ic(ic_closest)
```
  
  + **enhancement idea** - would be a nice feature if user could also supply an image instead of a date. The function then could find the closest image in the image collection to that date of the image supplied. Similar to functionalit of `find_closest()` function in this [example](https://developers.google.com/earth-engine/tutorials/community/histogram-matching). #za_ff4
- `Extra_minValue/Extra_maxValue()` - I think this function is only designed for `ee.Image` therefore some class assertions would be nice to avoid slow "ee.ee_exception.EEException: User memory limit exceeded" message if someone runs on anything else? #za_ff5
- `Extra_matchHistogram()` - probably a user error, but I had a hard time trying to reproduce the [Histogram Matching Example](https://developers.google.com/earth-engine/tutorials/community/histogram-matching) from Google. I think my issue likely relates to what was flagged in `#za_df1`  here is an attempt: #za_ff6
```{r}
library(rgee)
library(rgeeExtra)
library(tidyverse)
ee_Initialize()
extra_Initialize()

# make geom
geometry = ee$Geometry$Polygon(
  list(
    c(-155.97117211519446, 20.09006980142336),
    c(-155.97117211519446, 19.7821681268256),
    c(-155.73256280122962, 19.7821681268256),
    c(-155.73256280122962, 20.09006980142336)
  )
)
skysat = ee$Image('SKYSAT/GEN-A/PUBLIC/ORTHO/RGB/s01_20161020T214047Z')$clip(geometry)
landsat_ic = ee$ImageCollection('LANDSAT/LC08/C01/T1_SR')$
  filterBounds(geometry) %>% 
  # leaving as default -- this is probably wrong! but not super clear how to adjust
  # would be nice if vignettes showed good default presets for at least landsat & and sentinel 2
  ee$ImageCollection$Extra_preprocess() 

date_skysat <- ee_get_date_img(skysat)$time_start # if you run Extra_closest w/ this it aborts session
date_skysat_chr<- as.character(as_date(date_skysat))

landsat_ic_closest <- ee$ImageCollection$Extra_closest(landsat_ic ,date_skysat_chr,32,"day")

# would be nice if Extra_closest returns img in the above
landsat_img_closest <- landsat_ic_closest[[1]] # but alas we have to convert it

landsat_img_rgb <- landsat_img_closest$select(
  opt_selectors = c( "B4","B3","B2"),
  opt_names = c("R","G","B")
)

bands <- list("R"="R","G"="G","B"="B")
matched <- ee$Image$Extra_matchHistogram(skysat,landsat_img_rgb,  bands)

vis_params_refl = list('min'=0, 'max'= 0.25)
vis_params_dn = list('min'= 0, 'max'= 255)
aoi_center_pt <- ee$Geometry$Point(c(-155.79584,19.9986))
Map$centerObject(aoi_center_pt,13)

m1 <- Map$addLayer(landsat_img_rgb, visParams = vis_params_refl,"landsat ref")
m2 <- Map$addLayer(skysat,visParams= vis_params_dn, 'SkySat source')
m1|m2 # looks good
m3 <- Map$addLayer(matched,visParams= vis_params_refl, 'SkySat matched')
m4 <- Map$addLayer(matched,visParams= vis_params_dn, 'SkySat matched')
m1|m3 # matches is just black 
m1|m4 # matches is just black 
```

For a user like me who wrote the code below what are some tools we have from `{rgee}` and `{rgeeExtra}` to help troubleshoot and figure out what is going on? This is all I can think of right now but doesn't get me much further? #za_ff7
```{r}
# trying to trouble shoot - wrote this before I THINK i might have identified the problem
# in leaving the Extra_preprocess as default
rgeeExtra::ee_minValue(landsat_img_rgb) 
rgeeExtra::ee_maxValue(landsat_img_rgb) 
rgeeExtra::ee_minValue(skysat)
rgeeExtra::ee_maxValue(skysat) 

rgeeExtra::ee_minValue(matched) # all 0s
rgeeExtra::ee_maxValue(matched) # all 0s
```

- `Extra_cloudMask()` would be nice to have this for ics (`ee_ImageCollection_maskClouds`) #za_ff8

### Functionality - Code style

- in vignettes and examples you seem to prefer the `%>%` syntax for chaining together `{rgee}`/`{rgeeExtra}` processes like this:

```{r}
doi <- ee$ImageCollection("NASA/GPM_L3/IMERG_V06")[[1]] %>%
  ee$Image$Extra_getDOI()
```

rather than:
```{r}
doi <- ee$ImageCollection("NASA/GPM_L3/IMERG_V06")[[1]]$
  Extra_getDOI()

ic <- ee$ImageCollection("NASA/GPM_L3/IMERG_V06")

```

  + is this somehow best practices or you just think easier for R-users? I think the second looks cleaner (especially for longer chains) and might be easier for users migrating from the JavaScript code editor. #za_fc1
  + perhaps useful if it could be made clear somewhere in the documentation that you can do both styles and if there is a benefit to either of certain cases #za_fc2

- can remove leftover code
https://github.com/r-earthengine/rgeeExtra/blob/7ea67d064e56e9429971d64440d8424df68f6249/R/ops-math.R#L108-L110 #za_fc3
- In summary methods you rename the new band to "layer" no matter what operation( `summ_r$rename("layer")`)? Would it be safer to explicitly rename the band based on the band_name + Reducer name (i.e B1_max)? #za_fc4
- `raster.R` - why is this file called raster? name seems out of place? #za_fc5
- `ee_get()`- wonder if this name is a bit ambiguous considering all the other `get_*` functions there are in `rgee`,`GEE` and `rgeeExtra`... a couple ideas which make more sense to me filterIndex, filter_index, slice, subset, subset_index #za_fc6
- recommendation is use `<-` assignment over `->` -- #za_fc7 https://github.com/r-earthengine/rgeeExtra/blob/7ea67d064e56e9429971d64440d8424df68f6249/R/ee_gif.R#L187-L192

**Comment by maurolepore** [comment]
@zackarno, thanks a lot for your thorough review!

@csaybar please wait a bit for @nmorandeira's review then you can address both reviews.

**Comment by nmorandeira** [comment]
Thanks [@csaybar](https://github.com/csaybar) for developing this excellent package!, and thanks @maurolepore for the invitation to review it. Sorry for the delay, I'm out of my office with very limited internet access. I provide some comments, mainly related to the documentation. As I said to @maurolepore, my views are as an end-user, since I have no experience in developing a package. 

## Package Review
- [x]  As the reviewer I confirm that there are no [conflicts of interest](https://devguide.ropensci.org/policies.html#coi) for me to review this work (if you are unsure whether you are in conflict, please speak to your editor before starting your review).

## Documentation
The package includes all the following forms of documentation:

- [x] A statement of need: clearly stating problems the software is designed to solve and its target audience in README. 
_See below, I suggest adding more detail on this._

- [ ]  Installation instructions: for the development version of package and any non-standard dependencies in README.
_See my comment below. Installation instructions are insufficient._

- [x]  Vignette(s): demonstrating major functionality that runs successfully locally
_See my comments below._

- [ ]   Function Documentation: for all exported functions
_See my comments below._

- [x]   Examples: (that run successfully locally) for all exported functions
_See my comments below._

- [x]   Community guidelines: including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with URL, BugReports and Maintainer (which may be autogenerated via Authors@R).


## Functionality
- [ ] Installation: Installation succeeds as documented. 
I found problems initializing ee, see report below.

I haven't tested functionality, performance, automated tests and packaging guidelines.

Estimated hours spent reviewing: 10;  ~3 hours were spent in trying to solve the installation issues.

- [x] Should the author(s) deem it appropriate, I agree to be acknowledged as a package reviewer ("rev" role) in the package DESCRIPTION file.

## Review Comments

+ In the Readme of the repository and in the rgeeExtra webpage, section "Why rgeeExtra is needed?"[https://github.com/r-earthengine/rgeeExtra?tab=readme-ov-file#why-rgeeextra-is-needed-] I would add a few sentences explaining the table, for the user to quickly know the extra features of the package in comparison with rgee.

For example, _"While GEE's primary language is JavaScript, which diverges from R's structure, rgeeExtra serves as a bridge"_. Isn't this true for rgee? What are the additional advantages of using rgeeExtra? I now know the answer :), after using the package and reviewing the Articles, but It would be nice for the user to quickly decide if it's worth installing rgeeExtra or no.

+ The equation provided as example is confusing because is called “ndvi” but includes a _**2_. I would name the object _“ndvi_squared”_, or alternatively remove the _** 2_ from the equation.

+ Add to the Readme.MD in the repository a link to https://r-earthengine.com/rgeeExtra/index.html
Articles are a nice vignette! I would highlight them.

+ Citation: is there an error? In [this section](https://r-earthengine.com/rgeeExtra/index.html#share-the-love-%EF%B8%8F), it says
```

‘citation("rgeeExtra")
To cite **rgee** in publications use:
  
  C Aybar, Q Wu, L Bautista, R Yali and A Barja (2020) rgee: An R
  package for interacting with Google Earth Engine Journal of Open
  Source Software URL https://github.com/r-spatial/rgee/.

A BibTeX entry for LaTeX users is

@Article{,
  title = {rgee: An R package for interacting with Google Earth Engine},
  author = {Cesar Aybar and Quisheng Wu and Lesly Bautista and Roy Yali and Antony Barja},
  journal = {Journal of Open Source Software},
  year = {2020},
}
‘
```

In R, running citation(“rgeeExtra”) gives the corrected text “To cite **rgeeExtra** in publications use”, but the same reference by Aybar et al. (2020) refering to rgee.



+ **Installation.** This was one of the main issues I encountered while reviewing the package. I was not able to initialize ee, here follows my error (probably related to miniconda or Python?). I couldn’t solve this issue in my laptop. Instead, I decided to conduct the review on other PC were I had already set up my environment. My suggestion is to include a clear step-to-step instruction to install the package and to initialize ee. 

I followed the instructions provided here by Mauro: https://github.com/ropensci/software-review/issues/608#issuecomment-1755823040 . 
When running `rgee::ee_install() ` I get this error:

![image](https://github.com/ropensci/software-review/assets/6751201/cd46c9e6-8393-49c8-b91c-eb0613589437)

I'm successfully using gee with Python in this laptop, but probably my R + Python environment is not correctly configured.

+  **Function Documentation:** after reviewing the Articles, it's not clear for me if all the functions included in rgeeExtra are listed in the examples. It seems to me that there may be extra functions that I have still not explored. A vignette including the list of functions and its parameters would be appreciated.

## Articles
Here I provide some comments on the articles.

### Introduction
SAVI computation. I get this error:

```
Error in py_get_attr_impl(x, name, silent) : 
  AttributeError: type object 'Image' has no attribute 'spectralIndex'
```


Please check this error. I replaced “preprocess” with “Extra_preprocess”, and “spectralIndex” with “Extra_spectralIndex”. Next, I also replaced “geoviz_1” with “geoviz”. In this way I get no error, but I also don’t see any image in the visualizer. 

Besides, if it’s possible to compute an spectral index this way, I would add a comment in the Readme file: after giving the ndvi or squared ndvi equation as an example of how an user can define a new formula using the rgeeExtra syntax, I would link to this SAVI example (or an NDVI example) as an additional advantage of the package, i.e. providing an easy way to compute well known spectral indexes.

### Image
Add dependency:
`library(leaflet.extras2)
`
**1. Arithmetic Operations.** The screenshot image shows global coverage, but the code gives a plot with zoom 9. Be consistent so that the user doesn't think he/she has done something wrong.

**2. Mathematical functions.** Comment that you can visualize each NDVI transformation by clicking on the layer selector.

**4. Tasseled Cap Transformation.**
I suggest replacing the display by the most used RGB combination for tasseled cap (R = TCB, G = TCG, B = TCW).

A reference to the citation for the the tasseled cap transformation would be nice, as well as for the other indexes included in the articles (NDVI, SAVI, etc).

**6. Histogram Matching**
The visualization parameters should be different for each image. A possible solution (I have also added the source scene for comparison purposes):

```
visParams_L8 <- list(
  bands = c("B4","B3","B2"),
  min = 0.1, 
  max = 0.5, 
  gamma = 1.5
)

visParams_L7 <- list(
  bands = c("B3","B2","B1"),
  min = 0.1, 
  max = 0.5, 
  gamma = 1.5
)

# Display matched and target images on the map for comparison.
Map$setCenter(lon = -121.984, lat = 47.847, zoom = 14)
m0 <- Map$addLayer(source, visParams = visParams_L8, "source")
m1 <- Map$addLayer(matched, visParams = visParams_L8, "matched") 
 
m2 <- Map$addLayer(target, visParams = visParams_L7, "target")

m0 | m2

m1 | m2

```

**7. Cloud masking**
Please check the parameters. I don’t get the same display.
![image](https://github.com/ropensci/software-review/assets/6751201/341e5681-6aaf-4223-a7a4-6eda138fbb74)



**8. Automated Preprocessing**
Please check the parameters. I don’t get the same display.
![image](https://github.com/ropensci/software-review/assets/6751201/557ca1d3-d10f-427b-9b34-f2a3dc43fb3e)


**9. Summary Methods.**
Add visualization instructions.

**10. Minimum and Maximum values.**
Requires geojsonio, which also has several dependencies. Add: `library(geojsonio)`
Also, the text says _“These functions offer two modes: “Rectangle” for using the image’s footprint and “Points” for sampling over it”_. Can you add examples for these two modes?


## Image Collection
**8. Complete GIF Toolkit for Earth Engine.**
Excellent tool! Please add the dependency:
`library(magick)`

Also, save the gif file to the working directory (It was saved in a temporal folder with a random name, so it’s difficult to find it).

**11. Get the temporal nearest image.**
Excellent tool! I would move this example before example (8). Also, the title is in italics, it seems to be an error.
As an user, I would find much more useful to find the closest cloud-free image.

**Comment by maurolepore** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/608#issuecomment-1894601089 time 10

**Comment by maurolepore** [comment]
@ropensci-review-bot submit review https://github.com/ropensci/software-review/issues/608#issuecomment-1879736435 time 16

**Comment by maurolepore** [comment]
Thanks a lot @nmorandeira for your review! Your perspective is very valuable and really nicely complements that of @zackarno.

@csaybar you can now start addressing the reviews.

**Comment by csaybar** [comment]
Thanks, I will start with the revision.

**Comment by maurolepore** [comment]
> If the author mentions changes might take time, [apply the holding label via typing @ropensci-review-bot put on hold](https://devdevguide.netlify.app/softwarereview_policies#policiesreviewprocess). You’ll get a reminder every 90 days (in the issue) to check in with the package author(s).

--https://devdevguide.netlify.app/softwarereview_editor

--

Hi @csaybar, shall we put this submission on hold? 

It's totally fine to do so. The label helps us to better understand and manage the process.

**Comment by csaybar** [comment]
Hi everyone,

I sincerely apologize for my delayed response,

I had a bit of trouble properly solving all of your valid requests. As many of you know, rgee-extra is an extension of rgee. At the same time, rgee relies on the earthengine-api (Python). Therefore, updates to the Python client library directly impact rgee. Recently, the earthengine-api has undergone substantial changes concerning authentication (see updates [here](https://developers.google.com/earth-engine/changelog#python_api_v01383) and [here](https://developers.google.com/earth-engine/changelog#python_api_v01385)), which have break many of rgee’s functionalities.

After some internal discussions, we have decided to restructure rgee into a new package (to be named rgee2) to address installation & authentication problems. 

@JulioContrerasH is working part-time on developing rgee2, but we do not expect it to be ready before the end of August. We are committed to keeping rgee-extra as an extension of rgee2. However, we prefer to pause further development on the project until rgee2 first version is released. 

Thank you for your understanding, and sorry for the lengthy message. :(
Yes, @maurolepore, please can u move it to hold.

**Comment by maurolepore** [comment]
@csaybar, thanks for your efforts. I'll go ahead and put the submission on hold.

**Comment by maurolepore** [comment]
@ropensci-review-bot put on hold

**Comment by nmorandeira** [comment]
Thanks for the update @csaybar 
 Good luck with rgee2!

**Comment by maurolepore** [comment]
@csaybar 

Is there any change?

The bot is reminding us to review the holding status. This is not to put any pressure but just to ensure we don't forget. 

We'll get reminders every three months for a year, then we would need to move forward or close the submission.

**Comment by emilyriederer** [comment]
Hi @csaybar ! Just checking in again are there any updates here? 

To echo Mauro's point, at a point we may need to either move forward or close this out.

**Comment by csaybar** [comment]
Hi @emilyriederer 

Thank you so much for following up, I truly appreciate your understanding. I completely agree with the need to resolve this and regret that I haven’t been able to provide updates. Unfortunately, the changes required have proven more complex than anticipated, and with my energy stretched thin,  I no longer have the capacity to continue this work. We solved almost all the issues but the authentication problem persists.

Please feel free to close this out for now. My sincere apologies for the delays and the waste of time this has caused to everyone. I’m deeply grateful for all your patience and collaboration throughout this process.

Sorry everyone again and thank you for your kindness and flexibility.

**Comment by emilyriederer** [comment]
Hi @csaybar - thanks for the thoughtful note!

As  a matter of repo hygiene, I may close this for now, but please know we can re open at a later point in time if you want to complete the process.

Also, please let me say there is no reason to apologize. It sounds like the review process was hopefully still useful in its own way in surfacing some issues or ideas, and that's the more important outcome than a rubber stamp of approval. 

As I have not been involved in this review, I'm not qualified to assess what issues are mission-critical versus nice-to-haves, but also do realize review feedback is not a one-way street. If you think there are design or technical reasons to not implement some of the suggestions, that is also a constructive discussion we could have here.

Up to you, going forward! Closing for now but we're still happy to engage on a path forward if it feels right later.
